### PR TITLE
Add usage of int8-mixed-bf16 quantization with X86InductorQuantizer

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
## Description
Add usage of int8-mixed-bf16 quantization in X86InductorQuantizer.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @sekyondaMeta @svekars @carljparker @NicolasHug @kit1980 @subramen

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
cc @jgong5 @jerryzh168

File: prototype_source/pt2e_quant_ptq_x86_inductor.rst [hunk 1]
```diff
@@ -165,11 +165,25 @@ After we get the quantized model, we will further lower it to the inductor backe
...
+a Convolution or GEMM operator produces BFloat16 output data type instead of Float32 in the absence
+of a subsequent quantization node. Subsequently, the BFloat16 tensor seamlessly propagates through
+subsequent pointwise operators, effectively minimizing memory usage and potentially enhancing performance.
+
+::
 
-    # Running some benchmark
-    optimized_model(*example_inputs)
+    with torch.autocast(device_type="cpu", dtype=torch.bfloat16, enabled=True), torch.no_grad():
+        # Turn on Autocast to use int8-mixed-bf16 quantization
```
Comment (jerryzh168) [inline prototype_source/pt2e_quant_ptq_x86_inductor.rst:182]: can you also specify what is the quantized operator run here?

Comment (leslie-fang-intel) [reply prototype_source/pt2e_quant_ptq_x86_inductor.rst]: Thanks for the comment. Specified the input/output data type and computation precision for Conv/GEMM, quantizable operators and non-quantizable operators in this scenario. Please help to take a look again.

File: prototype_source/pt2e_quant_ptq_x86_inductor.rst [hunk 2]
```diff
@@ -165,11 +165,35 @@ After we get the quantized model, we will further lower it to the inductor backe
...
+
+        # Running some benchmark
+        optimized_model(*example_inputs)
 
-    # Running some benchmark
-    optimized_model(*example_inputs)
+In a more advanced scenario, int8-mixed-bf16 quantization comes into play. In this instance,
+a Convolution or GEMM operator produces BFloat16 output data type instead of Float32 in the absence
+of a subsequent quantization node. Subsequently, the BFloat16 tensor seamlessly propagates through
+subsequent pointwise operators, effectively minimizing memory usage and potentially enhancing performance.
```
Comment (jgong5) [inline prototype_source/pt2e_quant_ptq_x86_inductor.rst:177]: Shall we highlight the fact that the usage is as simple as wrapping the script under the bfloat16 autocast context?

**Review by jgong5 — APPROVED** [review]

File: prototype_source/pt2e_quant_ptq_x86_inductor.rst (hunk 2 above)
Comment (leslie-fang-intel) [reply prototype_source/pt2e_quant_ptq_x86_inductor.rst:177]: Thanks for the suggestion. Modified the description to highlight it.

File: prototype_source/pt2e_quant_ptq_x86_inductor.rst [hunk 3]
```diff
@@ -165,11 +165,37 @@ After we get the quantized model, we will further lower it to the inductor backe
...
+        # Turn on Autocast to use int8-mixed-bf16 quantization. After lowering into Inductor CPP Backend,
+        # For operators such as QConvolution and QLinear:
+        # * The input data type is consistently defined as int8, attributable to the presence of a pair
+            of quantization and dequantization nodes inserted at the input.
+        # * The computation precision remains at int8.
+        # * The output data type may vary, being either int8 or BFloat16, contingent on the presence
+        #   of a pair of quantization and dequantization nodes at the output.
+        # For non-quantizable pointwise operators, the data type will be inherited from the previous node,
+        # potentially resulting in a data type of BFloat16 in this scenario.
+        # For quantizable pointwise operators such as QMaxpool2D, it continues to operate with the int8
```
Comment (jerryzh168) [inline prototype_source/pt2e_quant_ptq_x86_inductor.rst:193]: meaning there are ops that are not affected by bfloat16 cast?

**Review by jerryzh168 — APPROVED** [review]
LG, thanks!

**Comment by svekars** [comment]
Can you please update the branch so we can test and merge this.

File: prototype_source/pt2e_quant_ptq_x86_inductor.rst (hunk 3 above)
Comment (leslie-fang-intel) [reply prototype_source/pt2e_quant_ptq_x86_inductor.rst:193]: Yes, for pointwise operators like QMaxpool2D, the behavior is identical to int8-mixed-fp32 when following a Conv/GEMM, as the cases in RN50

**Comment by leslie-fang-intel** [comment]
> Can you please update the branch so we can test and merge this.

@svekars rebased to the latest main branch. Please help to take a look again.
